pkg/skaffold/parser/config_test.go

```diff
@@ -1205,6 +1206,86 @@ requires:
+requires:
+  - path: doc1
+    configs: [cfg10]
+`}, {name: "cfg01", requiresStanza: ""}}},
+				{path: "doc1/skaffold.yaml", configs: []mockCfg{{name: "cfg10", requiresStanza: `
+requires:
+  - path: ../doc2
+    configs: [cfg21]
+`}, {name: "cfg11", requiresStanza: ""}}},
+				{path: "doc2/skaffold.yaml", configs: []mockCfg{{name: "cfg20", requiresStanza: ""}, {name: "cfg21", requiresStanza: ""}}},
+			},
+			expected: func(base string) []*latestV1.SkaffoldConfig {
+				return []*latestV1.SkaffoldConfig{
+					createCfg("cfg21", "pf0image21", filepath.Join(base, "doc2"), nil),
+					createCfg("cfg10", "pf0image10", filepath.Join(base, "doc1"), []latestV1.ConfigDependency{{Path: filepath.Join(base, "doc2"), Names: []string{"cfg21"}}}),
+					createCfg("cfg00", "pf0image00", ".", []latestV1.ConfigDependency{{Path: "doc1", Names: []string{"cfg10"}}}),
+					createCfg("cfg01", "pf0image01", ".", nil),
+				}
+			},
+		},
+		{
+			description:              "makePathsAbsolute false; recursively applied profiles",
+			makePathsAbsolute:        util.BoolPtr(false),
+			profiles:                 []string{"pf0"},
+			applyProfilesRecursively: true,
+			documents: []document{
+				{path: "skaffold.yaml", configs: []mockCfg{{name: "cfg00", requiresStanza: `
+requires:
+  - path: doc1
+    configs: [cfg10]
+`}, {name: "cfg01", requiresStanza: ""}}},
+				{path: "doc1/skaffold.yaml", configs: []mockCfg{{name: "cfg10", requiresStanza: `
+requires:
+  - path: ../doc2
+    configs: [cfg21]
+`}, {name: "cfg11", requiresStanza: ""}}},
+				{path: "doc2/skaffold.yaml", configs: []mockCfg{{name: "cfg20", requiresStanza: ""}, {name: "cfg21", requiresStanza: ""}}},
+			},
+			expected: func(base string) []*latestV1.SkaffoldConfig {
+				return []*latestV1.SkaffoldConfig{
+					createCfg("cfg21", "pf0image21", ".", nil),
+					createCfg("cfg10", "pf0image10", ".", []latestV1.ConfigDependency{{Path: "../doc2", Names: []string{"cfg21"}}}),
+					createCfg("cfg00", "pf0image00", ".", []latestV1.ConfigDependency{{Path: "doc1", Names: []string{"cfg10"}}}),
+					createCfg("cfg01", "pf0image01", ".", nil),
+				}
+			},
+		},
+		{
+			description:              "makePathsAbsolute true; recursively applied profiles",
+			makePathsAbsolute:        util.BoolPtr(true),
+			profiles:                 []string{"pf0"},
+			applyProfilesRecursively: true,
+			documents: []document{
+				{path: "skaffold.yaml", configs: []mockCfg{{name: "cfg00", requiresStanza: `
+requires:
+  - path: doc1
+    configs: [cfg10]
+`}, {name: "cfg01", requiresStanza: ""}}},
+				{path: "doc1/skaffold.yaml", configs: []mockCfg{{name: "cfg10", requiresStanza: `
+requires:
+  - path: ../doc2
+    configs: [cfg21]
+`}, {name: "cfg11", requiresStanza: ""}}},
+				{path: "doc2/skaffold.yaml", configs: []mockCfg{{name: "cfg20", requiresStanza: ""}, {name: "cfg21", requiresStanza: ""}}},
+			},
+			expected: func(base string) []*latestV1.SkaffoldConfig {
+				return []*latestV1.SkaffoldConfig{
```

Comment:
nit: config_test.go is a bit long atm with 56 test cases in `TestGetAllConfigs`.  From looking at it, this is likely justified as we want the full set of combos for each config option and the test framework is the same for all of them but wanted to point it out as it might be hard for newer users to read.

Reply:
I agree but didn't find any obvious way to simplify it. Do you have any ideas how this test can be better structured?